File: cogs/record_match.py
# cogs/cmd_record_match.py
import discord
from discord import app_commands
from discord.ext import commands
from utils.data_manager import load_data, save_data

@app_commands.command(name="record_match", description="Registra o resultado entre dois jogadores em um torneio")
@app_commands.choices(resultado=[
    app_commands.Choice(name="Vitória jogador1", value="1"),
    app_commands.Choice(name="Vitória jogador2", value="2"),
    app_commands.Choice(name="Empate", value="draw"),
])
async def record_match(interaction: discord.Interaction, player1: str, player2: str, resultado: app_commands.Choice[str], torneio: str = None):
    data = load_data()
    active = torneio or data.get("active")
    if not active:
        await interaction.response.send_message(" Não há torneio ativo nem foi especificado um.", ephemeral=True)
        return

    if active not in data["tournaments"]:
        await interaction.response.send_message(" Esse torneio não existe.", ephemeral=True)
        return

    tour = data["tournaments"][active]
    players = tour["players"]
    if player1 not in players or player2 not in players:
        await interaction.response.send_message(" Ambos os jogadores precisam estar cadastrados no torneio.", ephemeral=True)
        return

    p1 = players[player1]
    p2 = players[player2]
    p1["played"] += 1
    p2["played"] += 1

    if resultado.value == "1":
        p1["wins"] += 1
        p2["losses"] += 1
        p1["points"] += 3
    elif resultado.value == "2":
        p2["wins"] += 1
        p1["losses"] += 1
        p2["points"] += 3
    else:
        p1["draws"] += 1
        p2["draws"] += 1
        p1["points"] += 1
        p2["points"] += 1

    tour["matches"].append({"p1": player1, "p2": player2, "result": resultado.value})
    save_data(data)
    await interaction.response.send_message(f" Resultado registrado em **{active}**: **{player1}** vs **{player2}** → {resultado.name}")

async def setup(bot: commands.Bot):
    bot.tree.add_command(record_match)

File: src/commands/record-match.js
import {
  ApplicationIntegrationType,
  InteractionContextType,
  MessageFlags,
  SlashCommandBuilder,
} from "discord.js";

import { store } from "../data/json-store.js";

export const data = new SlashCommandBuilder()
  .setName("record_match")
  .setDescription("Registra o resultado entre dois jogadores em um torneio")
  .setIntegrationTypes(ApplicationIntegrationType.GuildInstall)
  .setContexts(InteractionContextType.Guild)
  .addStringOption((option) =>
    option
      .setName("player1")
      .setDescription("Primeiro jogador")
      .setMinLength(1)
      .setMaxLength(100)
      .setRequired(true),
  )
  .addStringOption((option) =>
    option
      .setName("player2")
      .setDescription("Segundo jogador")
      .setMinLength(1)
      .setMaxLength(100)
      .setRequired(true),
  )
  .addStringOption((option) =>
    option
      .setName("resultado")
      .setDescription("Resultado da partida")
      .setRequired(true)
      .addChoices(
        { name: "Vitória jogador1", value: "1" },
        { name: "Vitória jogador2", value: "2" },
        { name: "Empate", value: "draw" },
      ),
  )
  .addStringOption((option) =>
    option
      .setName("torneio")
      .setDescription("Nome do torneio")
      .setMinLength(1)
      .setMaxLength(100)
      .setRequired(false),
  );

const resultLabels = new Map([
  ["1", "Vitória jogador1"],
  ["2", "Vitória jogador2"],
  ["draw", "Empate"],
]);

export async function execute(interaction) {
  const player1 = interaction.options.getString("player1", true);
  const player2 = interaction.options.getString("player2", true);
  const result = interaction.options.getString("resultado", true);
  const requestedTournament = interaction.options.getString("torneio");
  let active;
  let error;

  await store.update((state) => {
    active = requestedTournament || state.active;
    if (!active) {
      error = " Não há torneio ativo nem foi especificado um.";
      return;
    }

    if (!Object.hasOwn(state.tournaments, active)) {
      error = " Esse torneio não existe.";
      return;
    }

    const tournament = state.tournaments[active];
    const players = tournament.players;
    if (!Object.hasOwn(players, player1) || !Object.hasOwn(players, player2)) {
      error = " Ambos os jogadores precisam estar cadastrados no torneio.";
      return;
    }

    if (player1 === player2) {
      error = " Um jogador não pode disputar uma partida contra si mesmo.";
      return;
    }

    const first = players[player1];
    const second = players[player2];
    first.played += 1;
    second.played += 1;

    if (result === "1") {
      first.wins += 1;
      second.losses += 1;
      first.points += 3;
    } else if (result === "2") {
      second.wins += 1;
      first.losses += 1;
      second.points += 3;
    } else {
      first.draws += 1;
      second.draws += 1;
      first.points += 1;
      second.points += 1;
    }

    tournament.matches.push({ p1: player1, p2: player2, result });
  });

  if (error) {
    await interaction.reply({ content: error, flags: MessageFlags.Ephemeral });
    return;
  }

  await interaction.reply(
    ` Resultado registrado em **${active}**: **${player1}** vs **${player2}** → ${resultLabels.get(result) ?? result}`,
  );
}
